src/main/java/com/enderio/core/common/transform/EnderCoreTransformer.java

```diff
@@ -28,14 +31,59 @@
 
 @MCVersion(value = "1.7.10")
 public class EnderCoreTransformer implements IClassTransformer {
+
+  // The EnderCore class cannot be referenced from the transformer,
+  // as it improperly triggers classloading of FML event classes.
+  // This is caused by its various event handlers. Since event handlers
+  // require an event class to be referenced in the method signature
+  // (in this case, a state event), the referenced event class is
+  // loaded along with the containing class of the method.
+  // Since EnderCore is referenced in the 'transform' method of this class,
+  // EnderCore and the aforementioned event classes are consequently referenced and classloaded too early.
+  //
+  // In order to be compatible with coremods such as Sponge, FML event
+  // classes cannot be classloaded until the game is starting. Specifically,
+  // they are being classloaded before FML is ready to actually start posting state events.
+  // This is significantly earlier than certain coremods such as Sponge expect,
+  // which prevents them from properly applying their own transformations.
+
+  // The solution is to copy the logger into this class, instead of referencing
+  // the static 'logger' field from EnderCore
+  public static final Logger logger = LogManager.getLogger("EnderCore");
+
+  // These classloader exclusions ensure that this transformer is not re-entrant =
+  // that is, it does not trigger any additional class transformation while
+  // it's transforming a class. While re-entrance is technically allowed,
+  // it can cause issues with more advanced transformers such as Mixin, and should
+  // therefore be avoided when at all possible.
+
+  // It is important to note that merely using a class in any way will not necessarily
+  // cause it be classloaded at the same time as the containing class. While any classes
+  // referenced from static fields, as well as any classes referenced from
+  // the signature of a method (both static and non-static), other uses such as referencing
+  // a class from within a method will not cause the referenced class to be immediately classloaded.
+  // Thus, even though AbstractConfigHandler references EnderCore several times,
+  // it is not necessary to add a transformer exclusion for EnderCore, since the
+  // reference occurs only within method bodies.
+
+  // The config packages are excluded as they are referenced at the start of
+  // the 'transform' method, in the usage of 'ConfigHandler.invisibleMode'.
+  // The 'Lang' class is referenced from a static field in AbstractConfigHandler,
+  // (see above paragraph) so it is excluded to.
+  static {
+    Launch.classLoader.addTransformerExclusion("com.enderio.core.common.config.");
```

Comment:
Typically you'd add these via an `@IFMLLoadingPlugin.TransformerExclusions` annotation